### PR TITLE
Add Navigator and more navigation pieces

### PR DESCRIPTION
**Pull request by autonome**

### CODE REVIEW

File: features/navigation.yml [hunk 1]
```diff
@@ -1,18 +1,25 @@
 name: Navigation API
 description: The `navigation` API provides mechanisms to initiate, intercept, and manage browser navigation actions.
```
Comment (ddbeck) [inline features/navigation.yml:2]:
```suggestion
description: The `navigation` API initiates, intercepts, or modifies browser navigation actions.
```

**Review by ddbeck — COMMENTED** [review]
One question and two suggestions for you. Looking pretty close though.

File: features/navigator.yml.dist [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  - api.Navigator.cookieEnabled
+
+  # baseline: false
+  # support:
+  #   chrome: "37"
+  #   chrome_android: "37"
+  #   edge: "15"
+  #   firefox: "48"
+  #   firefox_android: "48"
+  - api.Navigator.hardwareConcurrency
```
[the remark this answers is not in the document]
Comment (autonome) [reply features/navigator.yml.dist:90]: Hrm, I meant to bring this up in the original PR comment and forgot.

Caniuse has features for even more APIs that hang off here, eg `onLine`, `registerProtocolHandler`.

I was considering breaking them all out as features and adding a new group for them all, but wanted to get your take on it first before featurizing all of these bits, since it also leaves `navigator` oddly empty and "misc" feeling.

[the remark this answers is not in the document]
Comment (captainbrosset) [reply features/navigator.yml.dist:90]: It's true that `navigator` (also true for `window`) is mostly a global object that lets you access other features.

- `cookieEnabled` could go into a cookie feature, especially important once we start defining a feature for partitioned cookies.
- `onLine` might also be interesting as its own feature, together with the `online` and `offline` event listeners.
- I would definitely also want to split out the `registerProtocolHandler` things in their own feature. And probably group them with `html.manifest.protocol_handlers`.
- `userAgent` might also be interesting to consider on its own, especially because it would give us a way to add a description like "Not to be used anymore because UA Strings lie, and are hard to parse. Use client hints instead".
- ...

I'm having a real hard time finding properties and methods of the `navigator` global object that would make sense to keep here instead of in their own features.
I'm not completely opposed to having a navigator feature that acts as a catch-all for very old things no one cares about anymore, but a feature like this would be hard to name and describe too.

Other options include:
- Ignore `api.Navigator` altogether. It's a given, it will never not exist, and isn't, in itself, a feature.
- Or create a new feature called something like `global-browser-objects` which contains `api.Navigator`, `api.Window`, and other similar objects like `self`, `frames`, `globalThis`...

**Review by captainbrosset — DISMISSED** [review]

File: features/navigator.yml.dist (hunk 2 above)
[the remark this answers is not in the document]
Comment (autonome) [reply features/navigator.yml.dist:90]: Ok thanks, I'm flipping this back to draft, and will break out these as features. I agree that most of them stand alone.

>     * Ignore `api.Navigator` altogether. It's a given, it will never not exist, and isn't, in itself, a feature.

Hm, I am not a fan of leaving significantly visible parts of the platform out. That would just make things harder for newcomers. I'd prefer a symbolic placeholder feature w/ just that one single key, over leaving it out. I'll take that approach for now.

>     * Or create a new feature called something like `global-browser-objects` which contains `api.Navigator`, `api.Window`, and other similar objects like `self`, `frames`, `globalThis`...

I'm going w/ a mostly-empty Navigator for now. Maybe in the next meeting we could discuss an approach which brings these all together, whether feature or group.

[the remark this answers is not in the document]
Comment (autonome) [reply features/navigator.yml.dist]: Drafted this approach. Descriptions need more work, but here are some notes on feature composition approach. I don't think there's anything that's hard blocking here, but lmk if have suggestions for fixes or different approaches.

- `navigator` as a standalone feature... optionally we could make a group for all of the APIs hanging off these mega-objects, but if so we should do that consistently
- `navigator.cookiesEnabled` as a feature for now, because likely we end up w/ a group of cookie features due to implementation timeline and feature distinction
- `navigator.hardwareConcurrency` is currently in conflict because Caniuse has it all green w/ a note about Safari having it behind a flag, whereas BCD has it as baseline false due to no Safari support (i added a comment for now)
- `navigator.language` and related keys in a `Language` feature, but need to figure out whether to include text/audio/video track `language` properties, and also whether to include in the `intl` group (wondering how l10n fits there too...)
- `navigator.onLine` and friends - Doesn't quite match up exactly with Caniuse, since BCD reports supported while Caninuse considers the early Chrome/Safari partial implementation as not supported, just added a comment for now
- `navigator.pdfViewerEnabled` is all green on Caniuse but they have a separate feature for PDF viewing generally, which is marked not supported on Firefox/Chrome Android - i made a note on this feature, don't think we need to block on finding a magical fix
- `navigator.registerProtocolHandler` has a bunch of subkeys on BCD for supported schemes, etc - should figure out whether to include or not
- `navigator.userAgent` in a feature called `Client identification (user agent)` which is named to align w/ the spec and the direction towards client hinting over UA string

File: features/pdf-viewer.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+name: pdfViewerEnabled
+description: "The `navigator.pdfViewerEnabled` read-only property indicates whether the browser supports inline display of PDF files when navigating to them. If inline viewing is not supported the PDF is downloaded and may then be handled by an external application."
```
Comment (ddbeck) [inline features/pdf-viewer.yml:2]: The external application stuff is not in scope for us. Edits for brevity

```suggestion
description: The `navigator.pdfViewerEnabled` read-only property returns a Boolean for whether the browser navigates to and shows a PDF in the browser window or downloads the PDF.
```

File: features/pdf-viewer.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+spec: https://html.spec.whatwg.org/multipage/system-state.html#pdf-viewing-support
+# Caniuse has a feature for PDF viewing generally, which is marked not supported for
+# Chrome/Firefox Android, but also has this entry from the BCD key shown all green.
+# Not blocking on finding a resolution here, but should figure it out at some point.
+#caniuse: pdf-viewer
```
Comment (ddbeck) [inline features/pdf-viewer.yml:7]: I think the caniuse feature is about something else — whether the browser _can_ show PDFs inline, not whether its turned on and available. Can you remove all this and paste it an issue about it? I don't think it's really in-scope for us and would like to have that discussion elsewhere.

File: features/registerprotocolhandler.yml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+name: registerProtocolHandler
+description: "The `navigator.registerProtocolHandler()' method lets websites register their ability to open or handle particular address schemes (also known as protocols). For example, this API lets webmail sites open `mailto:` URLs, or VoIP sites open `tel:` URLs."
```
Comment (ddbeck) [inline features/registerprotocolhandler.yml:2]: I cannot remember the last time I heard the word "webmail." Also, some edits for brevity:

```suggestion
description: "The `navigator.registerProtocolHandler()' method declares a site's ability to handle an address scheme (also known as a protocol). For example, an email site can register to open `mailto:` URLs or a VoIP site to open `tel:` URLs."
```

**Review by ddbeck — CHANGES_REQUESTED** [review]

File: features/online.yml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+name: Online status
+description: "The `navigator.onLine` property returns a boolean value for the online status of the browser, with true meaning online and false meaning offline. The `online` and `offline` events communicate changes in this state, and are supported in windows and workers."
```
[the remark this answers is not in the document]
Comment (ddbeck) [reply features/online.yml:2]: Actually, I changed my mind, I misread the support data. Let's just not mention the worker/window distinction:

```suggestion
description: The `navigator.onLine` property returns a Boolean for whether the browser is connected to some network (though not necessarily the internet). The `online` and `offline` events fire when the connection state changes.
```

File: features/client-identification.yml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+name: Client identification (user agent)
```
[the remark this answers is not in the document]
Comment (autonome) [reply features/client-identification.yml:1]: I think @captainbrosset's idea was to use this as an opportunity to educate about not doing UA sniffing. But yeah, gotta call it that in order to be findable by the people whose behavior we want to change. This rename along with a suitable warning in the description should get the job done.

File: features/language.yml.dist [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+  # baseline: false
+  # support:
+  #   chrome: "4"
+  #   chrome_android: "18"
+  #   edge: "79"
+  #   firefox: "74"
+  #   firefox_android: "79"
+  #   safari_ios: "5"
+  - api.WorkerGlobalScope.languagechange_event
```
[the remark this answers is not in the document]
Comment (autonome) [reply features/language.yml.dist:75]: WTH. pinned it for now.

File: features/navigator.yml [hunk 9]
```diff
@@ -0,0 +1,6 @@
+name: Navigator
+description: "The `window.navigator` API represents the user agent and its state. It's also the generic global object under which many other APIs are located, such as `clipboard` or `wakeLock`."
+spec: https://html.spec.whatwg.org/multipage/system-state.html#the-navigator-object
+compat_features:
+  - api.Navigator
+  - api.Window.navigator
```
[the remark this answers is not in the document]
Comment (autonome) [reply features/navigator.yml:6]: I went with the latter approach, since hiding away such a prominent piece of the web in an unrelated feature doesn't seem right. Better to have it visible as a signal of future intent (even if explicitly saying it does nothing on its own).

File: features/pdf-viewer.yml (hunk 4 above)
Comment (autonome) [reply features/pdf-viewer.yml:7]: #2069

File: features/hardware-concurrency.yml [hunk 10]
```diff
@@ -0,0 +1,9 @@
+name: hardwareConcurrency
+description: "The `navigator.hardwareConcurrency` read-only property returns the number of logical processors available to run threads on the user's computer."
+spec: https://html.spec.whatwg.org/multipage/workers.html#navigator.hardwareconcurrency
+# Caniuse has this as green, with a note that Safari has it behind a flag.
+# BCD has this as baseline false due to lack of Safari support.
+caniuse: hardwareconcurrency
+compat_features:
+  - api.Navigator.hardwareConcurrency
+  - api.WorkerNavigator.hardwareConcurrency
```
[the remark this answers is not in the document]
Comment (ddbeck) [reply features/hardware-concurrency.yml]: I've sent https://github.com/mdn/browser-compat-data/pull/24815

[the remark this answers is not in the document]
Comment (ddbeck) [reply features/hardware-concurrency.yml]: I've also sent https://github.com/Fyrd/caniuse/pull/7190

File: features/language.yml.dist (hunk 8 above)
[the remark this answers is not in the document]
Comment (ddbeck) [reply features/language.yml.dist:79]: Sent https://github.com/mdn/browser-compat-data/pull/24816 to fix the api.WorkerGlobalScope.languagechange_event feature (it's not Baseline)

File: features/user-agent-sniffing.yml [hunk 11]
```diff
@@ -0,0 +1,6 @@
+name: User agent sniffing
+description: "The `navigator.userAgent` read-only property returns the user agent string for the current browser. Browser identification based on detecting the user agent string is unreliable and is not recommended."
```
Comment (ddbeck) [inline features/user-agent-sniffing.yml:2]: Optional. I think we can be a bit more specific about what we don't want people to do with this and what they ought to do instead. Maybe like this?

```suggestion
description: "The `navigator.userAgent` read-only property returns the user agent string for the current browser. Selectively showing content based on the user agent string is unreliable. Consider using feature detection instead."
```

**Review by ddbeck — APPROVED** [review]
One optional description amendment for you. Feel free to merge in any case.
